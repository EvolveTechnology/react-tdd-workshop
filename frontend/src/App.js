import React from "react";
import {
  Box,
  Button,
  Flex,
  Heading,
  Image,
  Text
} from "rebass/styled-components";

import Dialog from "@reach/dialog";

import { Checkout } from "components/Checkout";
import { ContributionCard } from "components/ContributionCard";
import { Footer } from "components/Footer";
import { Login } from "components/Login";
import { Navbar } from "components/Navbar";

import CoffeeCup from "assets/coffee_cup.png";
import CoffeeToGo from "assets/coffee_to_go.png";
import DebitCard from "assets/debit_card.png";
import ReactLogo from "assets/react_logo.png";

import "@reach/dialog/styles.css";

function App() {
  const [showDialog, setShowDialog] = React.useState(false);
  const open = () => setShowDialog(true);
  const close = () => setShowDialog(false);

  return (
    <Box
      width={["100%", "90%"]}
      maxWidth="1024px"
      margin="0 auto"
      bg="background"
    >
      <Navbar />
      <Flex px={2} py={4} alignItems="center" flexDirection="column">
        <Text m={2} fontSize={3} as="p" textAlign="center">
          Buy me a coffee is a way for you to support my work.
        </Text>
        <Text m={2} fontSize={3} as="p" textAlign="center">
          So far,{" "}
          <Text color="highlight" as="span" fontWeight="bold">
            3
          </Text>{" "}
          people have donated a total of{" "}
          <Text color="highlight" as="span" fontWeight="bold">
            4440
          </Text>{" "}
          cups of coffee.
        </Text>
        <Image
          src={CoffeeCup}
          my={1}
          sx={{
            width: ["40%", "30%", "25%", "20%", "15%"],
            cursor: "pointer"
          }}
          onClick={open}
        />

        <Button
          onClick={open}
          color="white"
          bg="highlight"
          fontSize={5}
          data-testid="donate-button"
        >
          Donate
        </Button>
      </Flex>

      <Dialog aria-label="Donate" isOpen={showDialog} onDismiss={close}>
        <Flex flexDirection="column">
          <Login />
          <Checkout onCancel={close} />
        </Flex>
      </Dialog>

      <Flex px={1} my={2} justifyContent="center">
        <Heading
          p={2}
          fontWeight="bold"
          fontSize={5}
          color="highlight"
          data-testid="contributors-title"
        >
          Contributors
        </Heading>
      </Flex>

      <Flex justifyContent="center" alignItems="center" flexDirection="column">
        <ContributionCard />
        <ContributionCard />
        <ContributionCard />
      </Flex>
      <Flex py={4} justifyContent="center">
        <Image src={DebitCard} />
        <Image src={CoffeeToGo} />
        <Image src={ReactLogo} width="64px" />
      </Flex>
      <Footer />
    </Box>
  );
}

export default App;
