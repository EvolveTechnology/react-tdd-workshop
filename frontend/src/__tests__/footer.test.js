import React from "react";

import App from "App";

import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

describe("Credit icon providers", () => {
  it("Mentions the icon providers", () => {
    expect(true).toBe(false);
  });

  it("Links to the icon providers", () => {
    expect(true).toBe(false);
  });
});

test("Credit style", () => {
  expect(true).toBe(false);
});
