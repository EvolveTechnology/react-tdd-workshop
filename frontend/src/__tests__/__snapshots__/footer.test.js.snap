// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Credit style 1`] = `"<div width=\\"90%\\" class=\\"sc-AxjAm hUBnld\\"><div color=\\"white\\" class=\\"sc-AxjAm sc-AxirZ cNfATY\\"><div font-weight=\\"bold\\" font-size=\\"2rem\\" class=\\"sc-AxjAm oczhH\\">Buy me a coffee</div><div class=\\"sc-AxjAm jeyKHA\\"></div><a href=\\"#!\\" font-size=\\"2\\" class=\\"sc-AxjAm kfAYCq\\">Log In</a></div><div class=\\"sc-AxjAm sc-AxirZ bIcLiT\\"><p font-size=\\"3\\" class=\\"sc-AxjAm hvXEqD\\">Buy me a coffee is a way for you to support my work.</p><p font-size=\\"3\\" class=\\"sc-AxjAm hvXEqD\\">So far, <span color=\\"highlight\\" font-weight=\\"bold\\" class=\\"sc-AxjAm doiciJ\\">3</span> people have donated a total of <span color=\\"highlight\\" font-weight=\\"bold\\" class=\\"sc-AxjAm doiciJ\\">4440</span> cups of coffee.</p><img src=\\"coffee_cup.png\\" class=\\"sc-AxjAm eMgQYg\\"><button color=\\"white\\" font-size=\\"5\\" class=\\"sc-AxjAm gGjrCw\\">Donate</button></div><div class=\\"sc-AxjAm sc-AxirZ fNVgX\\"><h2 font-weight=\\"bold\\" font-size=\\"2rem\\" color=\\"highlight\\" class=\\"sc-AxjAm byVCyg\\">Contributors</h2></div><div class=\\"sc-AxjAm sc-AxirZ dvHqUk\\"><div color=\\"highlight\\" width=\\"90%\\" class=\\"sc-AxjAm hEqCDb\\"><div class=\\"sc-AxjAm sc-AxirZ dbsPtB\\"><div class=\\"sc-AxjAm sc-AxirZ jwhmtK\\"><div class=\\"sc-AxjAm sc-AxirZ glDYrY\\"><h2 font-size=\\"3,4\\" class=\\"sc-AxjAm hIEflP\\">Anonymous</h2></div><p font-size=\\"3\\" style=\\"word-break: break-word;\\" class=\\"sc-AxjAm izSlj\\">Hi there! Keep up the awesome work!</p></div><div class=\\"sc-AxjAm sc-AxirZ etuibD\\"><div font-size=\\"5\\" class=\\"sc-AxjAm eNObaV\\"><p>+4</p><img src=\\"coffee_to_go.png\\" width=\\"44px\\" class=\\"sc-AxjAm eIMihv\\"></div></div></div></div><div color=\\"highlight\\" width=\\"90%\\" class=\\"sc-AxjAm hEqCDb\\"><div class=\\"sc-AxjAm sc-AxirZ dbsPtB\\"><div class=\\"sc-AxjAm sc-AxirZ jwhmtK\\"><div class=\\"sc-AxjAm sc-AxirZ glDYrY\\"><h2 font-size=\\"3,4\\" class=\\"sc-AxjAm hIEflP\\">Anonymous</h2></div><p font-size=\\"3\\" style=\\"word-break: break-word;\\" class=\\"sc-AxjAm izSlj\\">Hi there! Keep up the awesome work!</p></div><div class=\\"sc-AxjAm sc-AxirZ etuibD\\"><div font-size=\\"5\\" class=\\"sc-AxjAm eNObaV\\"><p>+40</p><img src=\\"coffee_to_go.png\\" width=\\"44px\\" class=\\"sc-AxjAm eIMihv\\"></div></div></div></div><div color=\\"highlight\\" width=\\"90%\\" class=\\"sc-AxjAm hEqCDb\\"><div class=\\"sc-AxjAm sc-AxirZ dbsPtB\\"><div class=\\"sc-AxjAm sc-AxirZ jwhmtK\\"><div class=\\"sc-AxjAm sc-AxirZ glDYrY\\"><h2 font-size=\\"3,4\\" class=\\"sc-AxjAm hIEflP\\">Anonymous</h2></div><p font-size=\\"3\\" style=\\"word-break: break-word;\\" class=\\"sc-AxjAm izSlj\\">Hi there! Keep up the awesome work!</p></div><div class=\\"sc-AxjAm sc-AxirZ etuibD\\"><div font-size=\\"5\\" class=\\"sc-AxjAm eNObaV\\"><p>+400</p><img src=\\"coffee_to_go.png\\" width=\\"44px\\" class=\\"sc-AxjAm eIMihv\\"></div></div></div></div></div><div class=\\"sc-AxjAm sc-AxirZ coteZM\\"><img src=\\"debit_card.png\\" class=\\"sc-AxjAm fFIAuE\\"><img src=\\"coffee_to_go.png\\" class=\\"sc-AxjAm fFIAuE\\"><img src=\\"react_logo.png\\" width=\\"64px\\" class=\\"sc-AxjAm cfWTjF\\"></div><div font-size=\\"2\\" class=\\"sc-AxjAm sc-AxirZ bsLIVh\\"><a href=\\"https://icons8.com/\\" rel=\\"noopener noreferrer\\" style=\\"text-decoration: underline;\\" class=\\"sc-AxjAm gWKrNW\\">Icons thanks to Icons8</a></div></div>"`;
